wemake_python_styleguide/visitors/ast/naming.py

```diff
@@ -60,11 +61,17 @@ def _check_name(self, node: ast.AST, name: str) -> None:
 
         if logical.does_contain_underscored_number(name):
             self.add_violation(UnderscoredNumberNameViolation(node, text=name))
+
         if logical.does_contain_consecutive_underscores(name):
             self.add_violation(
                 ConsecutiveUnderscoresInNameViolation(node, text=name),
             )
 
+        if logical.is_anonymous_variable(name):
```

Comment:
That's not how this should work.
We should forbid usage of the variable name, not its declaration. What do I mean by usage:
- `print(_)`
- `4 * _`
- `x = _`

What do I mean by declaration?
- `some, _ = call()`
- `(do_some() for _ in range(5))`

Reply:
I'm going through `green tree snakes` document to implement this rule. Meanwhile, let me know if there is a similar implementation in the repo.

Reply:
Nope, I do not think so. You need to get the context of `ast.Name`. It should not be `Load` for this variable.